qualtran/serialization/sympy.py

```diff
@@ -0,0 +1,222 @@
+        sympy_pb2.ConstSymbol.ImaginaryUnit: sympy.core.numbers.ImaginaryUnit(),
+    }
+    return enum_to_sympy[enum]
+
+
+def _get_const_symbolic_operand(expr: sympy.Expr) -> sympy_pb2.Parameter:
+    """
+    Helper function for serializing a symbolic constant from a sympy expression.
+
+    Currently supported symbolic constants are: pi, natural exponent, EulerGamma, sqrt(-1), and infinity.
+    """
+    if expr == sympy.pi:
+        return sympy_pb2.Parameter(const_symbol=sympy_pb2.ConstSymbol.Pi)
+    if expr == sympy.E:
+        return sympy_pb2.Parameter(const_symbol=sympy_pb2.ConstSymbol.E)
+    if expr == sympy.EulerGamma:
+        return sympy_pb2.Parameter(const_symbol=sympy_pb2.ConstSymbol.EulerGamma)
+    if isinstance(expr, sympy.core.numbers.Infinity):
+        return sympy_pb2.Parameter(const_symbol=sympy_pb2.ConstSymbol.Infinity)
+    if isinstance(expr, sympy.core.numbers.ImaginaryUnit):
+        return sympy_pb2.Parameter(const_symbol=sympy_pb2.ConstSymbol.ImaginaryUnit)
+    else:
+        raise NotImplementedError(f"Sympy expression {str(expr)} cannot be serialized.")
+
+
+def _get_sympy_operand(expr: Union[sympy.Expr, int, float]) -> sympy_pb2.Parameter:
+    """
+    Converts the input to a serializable sympy_pb2 Parameter.
+
+    A parameter represents a single, irreducable numeric entity such as a variable, constant, or an explicit number.
+    """
+
+    # Expression is a single, symbolic variable.
+    if isinstance(expr, sympy.core.symbol.Symbol):
+        return sympy_pb2.Parameter(symbol=str(expr))
+
+    # Expression is an integer
+    if issubclass(expr.__class__, sympy.core.numbers.Integer):
+        result = expr.numerator
+        if not isinstance(result, int):
+            raise NotImplementedError(f"Sympy expression {str(expr)} cannot be serialized.")
+        return sympy_pb2.Parameter(const_int=result)
+
+    # Expression cannot be broken down further, but is a constant.
+    if issubclass(expr.__class__, sympy.core.numbers.Number):
+        if isinstance(expr, sympy.core.numbers.Float):
+            return sympy_pb2.Parameter(const_float=float(expr))
+        if isinstance(expr, sympy.core.numbers.Rational):
+            numerator = _get_sympy_operand(expr.numerator)
+            denominator = _get_sympy_operand(expr.denominator)
+            fraction = sympy_pb2.Fraction(numerator=numerator, denominator=denominator)
+            return sympy_pb2.Parameter(const_rat=fraction)
+        else:
+            raise NotImplementedError(f"Sympy expression {str(expr)} cannot be serialized.")
+    if type(expr) == int:
+        return sympy_pb2.Parameter(const_int=expr)
+    if type(expr) == float:
+        return sympy_pb2.Parameter(const_float=expr)
+    else:
+        return _get_const_symbolic_operand(expr)
+
+
+def sympy_expr_to_proto(expr: sympy.Expr) -> sympy_pb2.Term:
+    """Serializes a sympy expression."""
+
+    function = _get_sympy_function_type(expr)
```

Comment:
Does this work when `expr` is a single variable / constant ? See suggested tests for more details

Reply:
Added tests for this case.